src/components/medication-dispense-review.component.tsx

```diff
@@ -1,202 +1,277 @@
-import React, { useState } from "react";
+import React from "react";
 import { CommonConfigProps, MedicationDispense } from "../types";
-import DispenseCard from "../components/dispense-card.component";
-import {
-  Select,
-  SelectItem,
-  TextArea,
-  FormGroup,
-  RadioButton,
-  Toggle,
-  FormLabel,
-  DataTableSkeleton,
-  Column,
-  ComboBox,
-  Grid,
-  NumberInput,
-} from "@carbon/react";
-import styles from "./medication-dispense-review.scss";
+import MedicationCard from "./medication-card-component";
+import { TextArea, Column, ComboBox, Grid, NumberInput } from "@carbon/react";
 import { useLayoutType } from "@openmrs/esm-framework";
 import { useTranslation } from "react-i18next";
+import { getMedication } from "../utils";
+import styles from "./medication-dispense-review.scss";
 
 interface MedicationDispenseReviewProps {
   medicationDispense: MedicationDispense;
+  index: number;
+  updateMedicationDispense: Function;
   drugDosingUnits: Array<CommonConfigProps>;
+  drugDispensingUnits: Array<CommonConfigProps>;
   drugRoutes: Array<CommonConfigProps>;
   orderFrequencies: Array<CommonConfigProps>;
 }
 
 const MedicationDispenseReview: React.FC<MedicationDispenseReviewProps> = ({
   medicationDispense,
+  index,
+  updateMedicationDispense,
   drugDosingUnits,
+  drugDispensingUnits,
   drugRoutes,
   orderFrequencies,
 }) => {
   const { t } = useTranslation();
   const isTablet = useLayoutType() === "tablet";
-  const [dispenseRequest, setDispenseRequest] = useState(medicationDispense);
```

Comment:
Creating the state here meant that the updates were not linked to the dispense-form component that was doing the actual submitting.  I changed this so that the medicationDispense and the updateMedicationDispensation  were passed down as props from the dispense form (see interface on the right)... I also had to pass down the index of the dispense so that it could update the proper dispense in the array.